tephra-hbase-compat-0.96/src/main/java/co/cask/tephra/hbase96/coprocessor/TransactionProcessor.java

```diff
@@ -173,11 +175,15 @@ public void preDelete(ObserverContext<RegionCoprocessorEnvironment> e, Delete de
     Put deleteMarkers = new Put(delete.getRow(), delete.getTimeStamp());
     for (byte[] family : delete.getFamilyCellMap().keySet()) {
       List<Cell> familyCells = delete.getFamilyCellMap().get(family);
-      int cellSize = familyCells.size();
-      for (int i = 0; i < cellSize; i++) {
-        Cell cell = familyCells.get(i);
-        deleteMarkers.add(CellUtil.cloneFamily(cell), CellUtil.cloneQualifier(cell), cell.getTimestamp(),
-            HConstants.EMPTY_BYTE_ARRAY);
+      if (isFamilyDelete(familyCells)) {
+        deleteMarkers.add(family, TxConstants.FAMILY_DELETE_QUALIFIER, HConstants.EMPTY_BYTE_ARRAY);
```

Comment:
Do we need to add to documentation, that byte[0] qualifier will be treated as delete tombstone, if users will write such cell?

Is there a way to distinguish between delete marker and user's byte[0] qualifier? Maybe we can add some special attribute to the cell?